alf/layers.py

```diff
@@ -3081,6 +3081,27 @@ def forward(self, input):
         return torch.lerp(input.detach(), input, self._scale)
 
 
+@alf.configurable
+class SummarizeGradient(ElementwiseLayerBase):
+    def __init__(self, name):
+        """A layer for summarizing the gradient of the input tensor.
+
+        Args:
+            name (str): used to describe the name of the summary, after the
+                tag 'tensor_gradient'.
+        """
+        super().__init__()
+
+        self._name = name
+
+    def forward(self, x):
+        # explicitly turn on gradient calculation in order to summarize gradient
+        x.requires_grad = True
```

Comment:
This can change the behavior of training. I suggest  do not summarize if x.requires_grad is false.

Reply:
Changed to only turn-on ``requires_grad`` for leaf nodes, which is a common use case when we want to summarize the gradient w.r.t. inputs.

Reply:
Now changed with a clone of the input as suggested.